### PR TITLE
Create 10.x branch for node < 6 users (NOT LTS)

### PR DESCRIPTION
**Pull request by sjelin**
***This PR does not indicate any LTS plans for node < 6***

We've had a few useful features come in since we broke off from node < 6, and it would probably make our lives easier to do another 10.x release with those features merged in.  This PR cherry-picks all the commits except for https://github.com/angular/webdriver-manager/commit/c16bf9053fc90e4b5e89ab867c514d0622ab0716, and also adds one new commit entitled `chore(npm version warnings): npm needs to be >= 3.0 to install appium >= 1.6` (which needs review).  I have verified that `npm test` works on ubuntu with node@4.7.0 and npm@3.10.10.

Our CI isn't configured for 10.x releases so ignore that.  Same with the CLA

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,5 +65,9 @@
     "run-sequence": "^1.1.5",
     "selenium-webdriver": "2.53.1",
     "typescript": "^2.1.1"
+  },
+  "engines": {
+    "node": ">=4",
+    "npm": ">=3"
```
Comment (sjelin) [inline package.json]: @cnishina is this a breaking change?  I can just remove this bit if it is

**Comment by cnishina** [comment]
Should we include these? These were written for ES6 promises:

- chore(downloader): replace workflow with a single request (#158) c0a65af16d26eca2770f2abc8c46479086e3985b
- feat(update): on update, write full binary paths to file (#140) 649fbe622b8b97d0ab30c98c7ed0458e9830f171
- feat(status): show the last downloaded version when using status (#177) e874341040ea1d3129701acc6498f156d944e5a3

**Comment by sjelin** [comment]
I don't see a good reason not too?  They run in node 4.7.0 at least

**Comment by sjelin** [comment]
Commits removed

**Comment by sjelin** [comment]
Ok, this time for real, no es6 promises

**Pull request merged**